src/dashboard/src/components/accounts/backends.py

```diff
@@ -42,6 +43,49 @@ class CustomOIDCBackend(OIDCAuthenticationBackend):
     Provide OpenID Connect authentication
     """
 
+    def get_settings(self, attr, *args):
+        if attr in [
+            "OIDC_RP_CLIENT_ID",
+            "OIDC_RP_CLIENT_SECRET",
+            "OIDC_OP_AUTHORIZATION_ENDPOINT",
+            "OIDC_OP_TOKEN_ENDPOINT",
+            "OIDC_OP_USER_ENDPOINT",
+            "OIDC_OP_JWKS_ENDPOINT",
+            "OIDC_OP_LOGOUT_ENDPOINT",
+        ]:
+            # Retrieve the request object stored in the instance.
+            request = getattr(self, "request", None)
+
+            if request:
+                provider_name = request.session.get("providername")
```

Comment:
Since this session value is the source of record for the current provider, my strategy was to rely on this variable completely rather than make local copies in the oidc derived classes. I thought this was simpler and more clear and avoids the problem of checking if the provider has changed, and comparing it to a stored provider name. This code will only be executed if one of these seven env vars are queried - any others will use the base class get_settings().

Is there a performance problem I am not accounting for with checking a session var?

Reply:
We plan to address this refactoring after adding some basic tests for the new components.

Reply:
I misread the code here. These were not iterations, but `if` conditions :facepalm:. Sorry for the noise.